sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -1170,8 +1170,8 @@ object GpuCast {
 
   private def escapeJsonString(cv: ColumnVector): ColumnVector = {
     withResource(cv) { _=>
-      val chars = Seq("\\", "\"")
-      val escaped = chars.map("\\" + _)
+      val chars = Seq("\r", "\n", "\\", "\"")
+      val escaped = Seq("\\r", "\\n", "\\\\", "\\\"")
```

Comment:
nit: less hard-coding
```suggestion
      val escaped = chars.map(org.apache.commons.text.StringEscapeUtils.escapeJava)
```